pyccl/tracers.py

```diff
@@ -585,3 +696,157 @@ def _check_returned_tracer(return_val):
     else:
         tr, _ = return_val
     return tr
+
+
+
+def get_MG_transfer_function(cosmo, z, k=None):
+    """ This function allows to obtain the function Sigma(z,k) (1 or 2D arrays)
+        for an array of redshifts (i.e. scale factors) coming from a redshift 
+        distribution (defined by the user) and a single value or an array of k 
+        specified by the user. We obtain then Sigma(z,k) as a 1D array for those z 
+        and k arrays and then convert it to a 2D array taking into consideration 
+        the given sizes of the arrays for z and k. The MG parameter array goes then
+        as a multiplicative factor within the MG transfer function. 
+        If k is not specified then only a 1D array for Sigma(a,k=0) is used.
+
+    Args:
+        cosmo (:class:`~pyccl.core.Cosmology`): cosmology object used to
+            transform redshifts into distances.
+        z (float or tuple of arrays): a single z value (for z_CMB for example) 
+            or a tuple of arrays (z, N(z)) giving the redshift distribution of the
+            objects. The units are arbitrary; N(z) will be normalized to unity.
+	k (float or array): a single k value or an array of k for which we 
+            calculate the MG parameter Sigma(a,k). For now, the k range should 
+            be limited to linear scales. 
+    """
+	
+    if isinstance(z, float):
+        a_single=1/(1+z)
+        a = np.linspace(a_single, 1, 100) 
+        # a_single is for example like for the CMB surface
+    else:
+        a = 1./(1+z)
+    a.sort()
+    # Scale-dependant MG case with an array of k 
+    if (isinstance(k, (list, np.ndarray))): 	
+        k.sort()
+	# computing MG factor array 
+        mgfac_1d = 1
+        mgfac_1d += Sig_MG(cosmo, a, k)	
+	# converting 1D MG factor to a 2D array, so it is compatible 
+        # with the transfer_ka input structure in MG_add.tracer and 
+        # add.tracer
+        mgfac_2d = mgfac_1d.reshape(len(a),-1,order='F')  
+    	# converting to ln(k) array 
+        lk = np.log(k) 
+    	# setting transfer_ka for this case
+        mg_transfer = (a, lk, mgfac_2d)  
+    # Scale-dependant MG case for a signle value of k 
+    elif (isinstance(k, float) or isinstance(k, int)): 		
+        mgfac_1d = 1
+        mgfac_1d += Sig_MG(cosmo, a, k)
+        # setting transfer function with an array of a and a value of k
+        mg_transfer = (a, mgfac_1d)  
+    # Scale-independent MG case (for k=None)
+    else:	
+        k = None
+        # computing MG factor
+        mgfac_1d = 1
+        mgfac_1d += Sig_MG(cosmo, a, k)
+        # setting transfer function with an array of a and k=None
+        mg_transfer = (a, mgfac_1d)  
+		
+    return mg_transfer
+
+
+    def MG_add_tracer(self, cosmo, kernel, mg_transfer, 
```

Comment:
There seem to be two instances of this function. I suspect this second one should disappear

Reply:
Done.